Input and Output of SYM, BIN, and RRC formats

robojay opened the pull request:
Adds output (m17-mod) and input (m17-demod) for the SYM, BIN, and RRC formats as described here: https://spec.m17project.org/appendix/file-formats

SYM - M17 symbols
BIN - packed M17 di-bits
RRC - RRC filtered and scaled M17 symbols

These are useful for debugging and developing other M17 implementations.

Input/output formats are selected by command line options.  See the --help on each command for details.

sp5wwp commented:
@mobilinkd can you consider this PR? We need this to move forward.

n1ai commented:
I have been using this code from Jay's fork for several weeks and it is is very useful.

mobilinkd commented:
Please fix #20, caused by your Windows patches first. Then we can look at adding new features.

mobilinkd commented:
This breaks backwards compatibility by unnecessarily changing command-line parameters. I will not accept that. Stable interfaces have real value.

Abraxas3d commented:
You're exactly right. Stable interfaces have real value and should be
protected.

For a project still in active development, with a small number of users
compared to the potential future user base, and I mean this in the most
respectful way possible, would you please reconsider this pull request?
These things appear to be in the specification.

If there's a better way to implement these functions without modifying
existing deployed code, then I know everyone would listen to your opinion
and advice.

Thank you for all the work you've done and published.

-Michelle Thompson




On Mon, Aug 1, 2022 at 12:36 AM Mobilinkd LLC ***@***.***>
wrote:

> This breaks backwards compatibility by unnecessarily changing command-line
> parameters. I will not accept that. Stable interfaces have real value.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mobilinkd/m17-cxx-demod/pull/21#issuecomment-1200527822>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABSNNZOBP3BBN4WXLY6RWOTVW4EXJANCNFSM5XOUXBUQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

robojay commented:
> Please fix #20, caused by your Windows patches first. Then we can look at adding new features.

Can I fix the README.md so that this requires C++20 ?  That will solve the problem.

> This breaks backwards compatibility by unnecessarily changing command-line parameters. I will not accept that. Stable interfaces have real value.

It wasn't an arbitrary change.  All the other command line options have a shortcut with the first letter of the option, *except* the one I changed.  I changed it to a) be consistent with all your other options, and b) make it consistent with the new options.

However, I will alter the new options to be all UPPER case shortcuts ('B', 'R', 'S'), and return the 'noise-blanker' shortcut to 'b', if that is acceptable.

robojay commented:
> > Please fix #20, caused by your Windows patches first. Then we can look at adding new features.
> 
> Can I fix the README.md so that this requires C++20 ? That will solve the problem.
> 
> > This breaks backwards compatibility by unnecessarily changing command-line parameters. I will not accept that. Stable interfaces have real value.
> 
> It wasn't an arbitrary change. All the other command line options have a shortcut with the first letter of the option, _except_ the one I changed. I changed it to a) be consistent with all your other options, and b) make it consistent with the new options.
> 
> However, I will alter the new options to be all UPPER case shortcuts ('B', 'R', 'S'), and return the 'noise-blanker' shortcut to 'b', if that is acceptable.

Ok, BERT is already 'B', so I will use 'x', 'r', and 's' for 'bin', 'rrc', and 'sym' respectively.
